bin/compile

```diff
@@ -151,6 +151,11 @@ if [[ "$YARN_2" == "true" ]]; then
     warn "There is an existing .yarnrc file that will not be used. All configurations are read from the .yarnrc.yml file." "https://devcenter.heroku.com/articles/migrating-to-yarn-2"
   fi
 
+  # YARN 2/3 does not like the YARN_PRODUCTION environment variable and will fail if it is set so
+  # we need to change the YARN_PRODUCTION environment variable to YARN_2_PRODUCTION
+  YARN_2_PRODUCTION=$YARN_PRODUCTION
+  unset YARN_PRODUCTION
```

Comment:
It would be my preference to warn here to let the user we're doing something unexpected. Essentially let them know we're doing something for them that they should do for themselves. 

SImilar to https://github.com/heroku/heroku-buildpack-ruby/blob/35f8db8137166f51f32865b1502432fde7e4d77c/lib/language_pack/ruby.rb#L332 (which could probably be enhanced). 

For example something like:

```
Warning: You have set YARN_PRODUCTION environment variable which is not compatible with Yarn 2

We have moved the value over to the YARN_2_PRODUCTION environment variable. Please update your heroku config vars to remove YARN_PRODUCTION and set YARN_2_PRODUCTION.

<link if you wanna be REALLY killing it, but I don't think it's necessary>
```